lib/offline/SQLiteWrapper.hpp

```diff
@@ -204,18 +204,34 @@ namespace MAT_NS_BEGIN {
     class SqliteDB {
         std::mutex m_lock;
     public:
-        SqliteDB(bool skipInitAndShutdown)
+        SqliteDB(bool skipInitAndShutdown,
+                 std::mutex* initAndShutdownLock = nullptr,
+                 int* instanceCount = nullptr)
             : m_db(nullptr),
-            m_skipInitAndShutdown(skipInitAndShutdown)
+              m_skipInitAndShutdown(skipInitAndShutdown),
+              m_initAndShutdownLock(initAndShutdownLock),
+              m_instanceCount(instanceCount)
         {
         }
 
         bool initialize(std::string const& filename, bool deletePrevious, size_t maxHeapLimit = 0)
         {
-            int result;
+            int result = SQLITE_OK;
 
             if (!m_skipInitAndShutdown) {
-                result = g_sqlite3Proxy->sqlite3_initialize();
+                if (m_initAndShutdownLock && m_instanceCount)
+                {
+                    LOCKGUARD(*m_initAndShutdownLock);
+                    *m_instanceCount = *m_instanceCount > 0 ? *m_instanceCount + 1 : 1;
```

Comment:
nit: this construct looks unusual, at least to my eye 😄

Reply:
I agree it wasn't very clear, updated the logic and added a check to not increment the count unless the initialization succeeds.